### PR TITLE
Degenerate Base Expansion addition

### PR DESCRIPTION
**Pull request by DarianHole**
Addressing Issue #60:

Added:
- Kmer expansion code
- Check for number of kmers produced
- Unit tests
- default allowed kmers is 100,000

### CODE REVIEW

**Comment by DarianHole** [comment]
Previous bugfix should probably be implemented before this one is

**Comment by DarianHole** [comment]
I have another prediction method that can be done before the kmers are created which is probably preferred although again it has a nested for loop: 
```python
for kmer in kmers:
    value = 1
    for char in kmer:
        length_key = len(d[char])
        value = value * length_key
    kmer_number = kmer_number + value
```

**Comment by DarianHole** [comment]
I currently have the kmer check before the automaton creation but isolated so it parses the scheme fasta twice. This is due to the way I have it keeping track of the number of kmers. 

I assume I should try to put the check into the automaton creation instead to lower run times and a better code. I'll see if I can create a solution for this

**Comment by DarianHole** [comment]
I'll throw the code here, let me know which way is wanted. This method can check in automaton so we don't have to parse the scheme twice. It exits as soon as it hits the limit however I could put the error message after in the automaton function instead of the max_kmer function.

```python
def check_total_kmers(kmer, total):
    kmer_number = 1
    for char in kmer:
        length_key = len(d[char])
        kmer_number = kmer_number * length_key
    total = total + kmer_number
    if total > 150:
        return logging.error('Did it work?')
    return total


total = 0
for kmer in kmers:
    total = check_total_kmers(kmer, total)
```

File: bio_hansel/aho_corasick/__init__.py [hunk 1]
```diff
@@ -1,12 +1,60 @@
...
+    for header, sequence in parse_fasta(scheme_fasta):
+        value = 1
+        for char in sequence:
+            length_key = len(bases_dict[char])
+            value = value * length_key
+        kmer_number = kmer_number + value
+    if kmer_number*2 > subtyping_params.max_degenerate_kmers:
+        return logging.error(
+            '''
+    Your current scheme contains "{}" kmers which is over the reccomended number of "{}".
```
Comment (Takadonet) [inline bio_hansel/aho_corasick/__init__.py]: Maybe change wording that the 'reccomended number' is either the default value OR what was passed to max-degenerate-kmers' flag. 

Also typo  'reccomended'

**Pull request merged**